Fix broken reservation link on single cars

aceDavon opened the pull request:
## In this PR, I have implemented the following tasks: :writing_hand: 

- [ ] Fix Broken reservation link on single cars page

aceDavon committed:
Fix broken reservation link on single cars

HabibPayenda commented on src/components/SingleCarPage.js:
```diff
@@ -20,59 +20,45 @@ export default function SingleCarPage() {
         </ul>
         <div className={styles.btnContainer}>
-          <button type="button" onClick={() => console.log('Reserve')} className={styles.btn}>
+          <Link to="/reserve" className={styles.btn}>
```
You have put in a great deal of work but as the requirements say we should pass the selected car to the reservations screen and it should be preselected there.

HabibPayenda reviewed (changes requested):
## Changes Required:
Hello @aceDavon 
 You have put in a great deal of work but as the requirements say we should pass the selected car to the reservations screen and it should be preselected there.

## To Highlight: :green_circle: 

Good/Professional git workflow :heavy_check_mark: 
Professional commit messages :heavy_check_mark: 
Linters passing 

Happy Coding! :fireworks:

aceDavon committed:
Pass an iterated car id as props on link

Pass a selected car id as props on the link button to the reservation
page

aceDavon committed:
Auto select a particuular car using Id

Destructure the passed ID from the previous page, set it to local state,
and select the Car's ID automatcally on the dropdown

aceDavon committed:
Add component test for the Homepage

aceDavon committed:
Add component test for cars details page

aceDavon committed:
Add test for the cars reservation page

aceDavon committed:
Implement the Cars delet component testing

aceDavon committed:
Fix linters error

iKrystyan reviewed (approved):
## Status: Approved :champagne: :partying_face: 
Hello @aceDavon 
Congratulations! You have put in a great deal of work and I must say there's nothing else to say other than __'It is time to merge it!'__

## To Highlight: :green_circle: 

Good/Professional git workflow :heavy_check_mark: 
Professional commit messages :heavy_check_mark: 
Linters passing :heavy_check_mark: 

Congratulations again and Happy Coding! :fireworks:
